src/cert_context.rs

```diff
@@ -107,6 +107,54 @@ impl CertContext {
         self.get_encoded_bytes()
     }
 
+    /// Deletes the cert's associated key container for both CAPI and CNG keys
+    pub fn del_key_container(&mut self) -> io::Result<()>{
+        unsafe {
+            // get CERT_KEY_PROV_INFO_PROP_ID and cast to key_prov
+            let bytes = self.get_bytes(Cryptography::CERT_KEY_PROV_INFO_PROP_ID)?;
+            assert!(bytes.len() <= u32::max_value() as usize);
+            let key_prov = bytes.as_ptr() as *const Cryptography::CRYPT_KEY_PROV_INFO;
+            
+            if (*key_prov).dwKeySpec == 0 { // CNG Key
+                let mut prov_handle = Cryptography::NCRYPT_PROV_HANDLE::default();
+                let mut key_handle = Cryptography::HCRYPTPROV_OR_NCRYPT_KEY_HANDLE::default();
+                
+                let ret = Cryptography::NCryptOpenStorageProvider(&mut prov_handle, (*key_prov).pwszProvName, 0);
+                if ret != 0 {
```

Comment:
> if ret != 0 { [](http://example.com/codeflow?start=15&length=14)

Add following so prov_handle is freed.

if ret == 0 {
ncrypt_prov = NcryptProv::from_inner(handle);
} else {